platform_channels/lib/src/pet_list_screen.dart

```diff
@@ -25,7 +24,7 @@ class _PetListScreenState extends State<PetListScreen> {
     BasicMessageChannel('stringCodecDemo', StringCodec())
         .setMessageHandler((message) async {
       if (message == null) {
-        scaffoldKey.currentState.showSnackBar(
+        ScaffoldMessenger.of(context).showSnackBar(
```

Comment:
Hello @AyushBherwani1998   I have added  `ScaffoldMessenger.of(context)` instead of using`scaffoldKey.currentState` though it's working fine  because after  wrapping `scaffold`   with the below snippet  the `Snackbar` is not showing

```
final GlobalKey<ScaffoldMessengerState> scaffoldMessengerKey = GlobalKey<ScaffoldMessengerState>();
ScaffoldMessenger(
  key: scaffoldMessengerKey,
  child: ...
)
```
can you please tell me it's right or not?

Thank you! :)

Reply:
If after wrapping Scaffold with ScaffoldMessenger and giving scaffoldMessengerKey, it didn't work, then you should try to reproduce a minimal example and file a bug in the Flutter SDK repo.

Reply:
> The snippet you have shown looks alright, If after wrapping Scaffold with ScaffoldMessenger and giving scaffoldMessengerKey, it didn't work, then you should try to reproduce a minimal example and file a bug in the Flutter SDK repo.

Sure @AyushBherwani1998.

Reply:
Sorry for pointing out late, we already have `showSnackBar` function which you can use.

Reply:
Sure @AyushBherwani1998  but while looking through this [document](https://master-api.flutter.dev/flutter/material/ScaffoldMessenger-class.html)  the `ScaffoldMessenger` should have a descendant `Scaffold` so  I would expect `ScaffoldMessenger` to work under `Scaffold` not outside of it.

```
When the ScaffoldMessenger has nested Scaffold descendants, the ScaffoldMessenger will only present a SnackBar to the root Scaffold of the subtree of Scaffolds. In order to show SnackBars in the inner, nested Scaffolds, set a new scope for your SnackBars by instantiating a new ScaffoldMessenger in between the levels of nesting.
```
**This doesn't work**


<details>
  <summary>
complete runnable code sample
  </summary>

   
```dart
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Material App',
      theme: ThemeData.dark(),
      home: Home(),
    );
  }
}

class Home extends StatefulWidget {
  @override
  _HomeState createState() => _HomeState();
}

class _HomeState extends State<Home> {
  final GlobalKey<ScaffoldMessengerState> scaffoldMessengerKey =
      GlobalKey<ScaffoldMessengerState>();

  @override
  void initState() {
    Future.delayed(
      Duration(milliseconds: 250),
      () {
        ScaffoldMessenger.of(context).showSnackBar(
          SnackBar(content: Text("Hello Flutter Dev !")),
        );
      },
    );
    super.initState();
  }

  @override
  Widget build(BuildContext context) {
    return ScaffoldMessenger(
      key: scaffoldMessengerKey,
      child: Scaffold(
        appBar: AppBar(
          title: Text('Material App Bar'),
        ),
        body: Center(
          child: Container(
            child: Text('Hello World'),
          ),
        ),
        floatingActionButton: FloatingActionButton(
          child: Icon(Icons.add),
          onPressed: () {},
        ),
      ),
    );
  }
}
```




























</details>




---

**This works, make sense to me**


<details>
  <summary>
complete runnable code sample
  </summary>

   
```dart

import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Material App',
      theme: ThemeData.dark(),
      home: Home(),
    );
  }
}

class Home extends StatefulWidget {
  @override
  _HomeState createState() => _HomeState();
}

class _HomeState extends State<Home> {
  final GlobalKey<ScaffoldMessengerState> scaffoldMessengerKey =
      GlobalKey<ScaffoldMessengerState>();

  @override
  void initState() {
    Future.delayed(
      Duration(milliseconds: 250),
      () {
        ScaffoldMessenger.of(context).showSnackBar(
          SnackBar(content: Text("Hello Flutter Dev !")),
        );
      },
    );
    super.initState();
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Material App Bar'),
      ),
      body: ScaffoldMessenger(
        key: scaffoldMessengerKey,
        child: Center(
          child: Container(
            child: Text('Hello World'),
          ),
        ),
      ),
      floatingActionButton: FloatingActionButton(
        child: Icon(Icons.add),
        onPressed: () {},
      ),
    );
  }
}
```




























</details>




---

<details>
  <summary>
  flutter doctor -v
</summary>

```bash

[√] Flutter (Channel master, 1.26.0-18.0.pre.49, on Microsoft Windows [Version 10.0.19041.746], locale en-IN)
    • Flutter version 1.26.0-18.0.pre.49 at E:\Opensource work\software\flutter
    • Framework revision 29e604e241 (3 days ago), 2021-01-28 00:04:03 -0800
    • Engine revision 68cd4c42b5
    • Dart version 2.12.0 (build 2.12.0-267.0.dev)

[!] Android toolchain - develop for Android devices (Android SDK version 30.0.3) 
    • Android SDK at C:\Users\DELL\AppData\Local\Android\sdk
    • Platform android-30, build-tools 30.0.3
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java       
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    X Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/windows#android-setup for more details.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[!] Visual Studio - develop for Windows (Visual Studio Community 2019 16.7.7)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.7.30621.155
    X Visual Studio is missing necessary components. Please re-run the Visual Studio installer for the "Desktop development with C++" workload, and include these components:    
        MSVC v142 - VS 2019 C++ x64/x86 build tools
         - If there are multiple build tool versions available, install the latest
        C++ CMake tools for Windows
        Windows 10 SDK

[√] Android Studio (version 4.0)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 52.1.1
    • Dart plugin version 201.9317
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[√] IntelliJ IDEA Community Edition (version 2020.3)
    • IntelliJ at C:\Program Files\JetBrains\IntelliJ IDEA Community Edition 2020.3
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart

[√] VS Code (version 1.52.1)
    • VS Code at C:\Users\DELL\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.18.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19041.746]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 88.0.4324.104
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 88.0.705.56

! Doctor found issues in 2 categories.
```

</details>

Reply:
Sorry, I think there's some miss communication. 

In previous commit when you used scaffoldMessengerKey.currentState.showSnackBar, the scaffoldMessengerKey was passed to Scaffold instead of ScaffoldMessenger, which was not correct. When you have passed the key to Scaffold, you can not access the ScaffoldMessengerState, hence scaffoldMessengerKey.currentState will be null, leading to an error.

That's why I said 

> Also, JFYI, this will given an error about currentState being null. In order to make it work, you need to wrap Scaffold with ScaffoldMessenger and pass the scaffoldMessengerKey in ScaffoldMessenger.key

The two examples you have shown are working correctly. The first example is not working because, by default, ScaffoldMessenger widget is introduced by the MaterialApp at the top of your application widget tree, in which all descendant Scaffolds register to receive SnackBars. After wrapping the Scaffold with ScaffoldMessenger, you are creating your own scope of ScaffoldMessenger, breaking the scope of root ScaffoldMessenger. 

In the first example when you call `ScaffoldMessenger.of(context)` you will access the state of root ScaffoldMessenger instead of local ScaffoldMessenger because of InheritedWidget. Since there are no Scaffold attached to root ScaffoldMessenger, it will not show any Snackbar.

The second example works, because ScaffoldMessnger is descendant of the Scaffold, which allows it to maintain the scope with the root ScaffoldMessenger.

In order to make the first example work, you have two options.
1. Either use key to access the currentState of local ScaffoldMessenger like below
```dart
scaffoldMessengerKey.currentState.showSnackBar(
          SnackBar(content: Text("Hello Flutter Dev !")),
        );
```

2. Or you can simply remove the ScaffoldMessenger allowing root ScaffoldMessenger to access the Scaffold and show Snackbar.

Reply:
Ook got it @AyushBherwani1998  now I could find why the first example was not showing `snackbar ` by wrapping scaffold with `ScaffoldMessanger` with `ScaffoldMessanger.key` as I was accessing the state of root `ScaffoldMessenger` instead of local `ScaffoldMessenger`.

sorry @AyushBherwani1998  for the miscommunication as we want to remove the use of  `currentState`  i think we are good to go with the existing PR.

Thank you! :)

Reply:
No need to apologize, there's only a single change I would suggest. On line 93 we already have a showSnackBar function which you can use if the message is null on line 23.